### DISCUSSION TITLE
Welcome to okama Discussions!

### DISCUSSION THREAD

**Discussion by chilango74** [mbk-dev/okama#3] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by garej** [comment]
Hi, how to add a point of the portfolio (as a whole 'asset') on a scatter with assets and efficient frontier? 
I did not find an option in the code. If one is supposed to add it manually, what are recommended attributes for risk and return (i.e. if I use `kind = "cagr"`)?

**Reply by chilango74** [reply]
There is no possibility to plot portfolio risk/return point in Plots class. Actually, I don't think it's required. It's very easy to add point by `.scatter`. If you draw the rest of assets with `kind="cagr"` then portfolio point should use the same metric: `.scatter(ok.Portfolio().risk_annual, ok.Portfolio().cagr)`.

**Reply by garej** [reply]
Thank you for the response. Why do you think it is not required? If you produced a bunch of montecarlo points (random portfolios), it seems reasonable to know where your actual point (portfolio) is located. Or am I missing something?

Another example is to compare several alternative portfolios - i.e. if I want to add a couple of new Tickers and see where my portfolio would drift.

**Reply by chilango74** [reply]
I meant that it's not required in the package because it's easy to do through matplotlib directly.

But we are considering another idea. It would be interesting to include *Portfolio* as an asset in the *AssetList*. If we implement it, portfolio could be compared with another benchmarks. And `plot_assets()` would also work with such portfolios.

**Comment by garej** [comment]
Another question is about labels on the scatter `plot_assets()`. Are you going to add a possibility to avoid labels overlapping if a portfolio is extensive?

**Reply by chilango74** [reply]
okama is not intended to be a visualization package. It has a simple Plots class. But it’s like a number of shortcuts to frequently used matplotlib command combinations. 

I don’t know an easy way to avoid labels overlapping. If you can suggest a solution, I would gladly introduce it in the code.
You can try to fine tune labels manually by changing their coordinates:
```python
        for label, x, y in zip(asset_labels, risks, returns):
            ax.annotate(label,  # this is the text
                        (x * m, y * m),  # this is the point to label
                        textcoords="offset points",  # how to position the text
                        xytext=(0, 10),  # distance from text to points (x,y)
                        ha='center')  # horizontal alignment can be left, right or center
```

**Reply by garej** [reply]
I see. Well, one option could be to use a wrapper like in this package https://github.com/Phlya/adjustText, but it would require to use `text` instead of `annotation`.  Anyway, I see your point about approach to plotting .

**Reply by chilango74** [reply]
Thanks. It looks lie an interesting package for labeling. I'll try to use it in okama.

**Comment by garej** [comment]
What about bonds? Are you going to add them in repository?

**Reply by chilango74** [reply]
We use **adjusted close** historical data to calculate risk and return metrics in the package. However, for bonds there is no such data. Therefore, it's not possible to include their historical data in the database.

You can use bonds ETFs as a proxy.

**Comment by garej** [comment]
Hi there, 
I did not find the way to see a time series of prices for an asset. Is it hidden or am I just missing something simple?

**Reply by chilango74** [reply]
Hello!
Price is available as a real time float only. We have close time series in the database. However, most asset risk/return metrics use **adjusted close** time series (close prices are adjusted for splits and dividends). Only dividend yield uses close. 
You can still get simple close daily or monthly time series through .get_close method in QueryData class.
Example:
``` python
Import okama as ok
ok.QueryData.get_close(symbol='SPY.US', first_date='2000-01', last_date='2010-01', period='D')
```
I think to introduce close time series as an asset property in closest releases too …

**Comment by garej** [comment]
How is solved the problem of getting recommended **shares for all assets** in the porfolio subject to required combination of "risk-return" for the whole portfolio? 
I see a transition map, but how do I know which return each transition combination shows? Is there a way to redraw it given arbitrary return?

**Reply by chilango74** [reply]
You are right. **Transition map** could be used to guess the allocation weights. However, the final decision is always taken by the investor. Portfolios laying on the Efficient Frontier (EF) are optimized for the historical data. But in real life we usually do not take such allocations. Transition map and the EF just help to limit the choices.

>To use the E-V rule in the selection of securities we must have procedures for finding reasonable [estimates of expected return and >volatility]. These procedures, I believe, should combine statistical techniques and the judgment of practical men. My feeling is that the >statistical computations should be used to arrive at a tentative set of [mean and volatility]. Judgment should then be used in increasing >or decreasing some of these [mean and volatility inputs] on the basis of factors or nuances not taken into account by the formal >computations…”
>
>  Harry M. Markowitz,  Portfolio Selection 1952
